src/qibolab/pulses.py

```diff
@@ -191,10 +190,9 @@ def envelope_waveform_i(self) -> Waveform:
             waveform = Waveform(self.pulse.amplitude * np.ones(num_samples))
             waveform.serial = f"Envelope_Waveform_I(num_samples = {num_samples}, amplitude = {format(self.pulse.amplitude, '.6f').rstrip('0').rstrip('.')}, shape = {repr(self)})"
             return waveform
-        else:
-            raise Exception(
-                "PulseShape attribute pulse must be initialised in order to be able to generate pulse waveforms"
-            )
+        raise RuntimeError(
+            "PulseShape attribute pulse must be initialised in order to be able to generate pulse waveforms"
+        )
```

Comment:
This string with this error is repeated many many times. Could make a custom exception subclassing `RuntimeError`, and embedding that string?

```py
class ShapeInitError(RuntimeError):
    default_msg = "PulseShape attribute pulse must be initialised in order to be able to generate pulse waveforms"
	
    def __init__(self, msg=None, *args):
        if msg is None:
            msg = self.default_msg
        super().__init__(msg, *args)
```

Such that you can just:
```suggestion
        raise ShapeInitError
```
here and below (unless you have a different message to pass).